Pod/Classes/Core/ARDynamicScreenDimensions.m

```diff
@@ -64,17 +64,18 @@ + (BOOL)requiresMainQueueSetup
 
 - (void)orientationChanged:(NSNotification *)note
 {
-  self.currentDimensions = getScreenDimensions();
   __weak ARDynamicScreenDimensions *wself = self;
 
-  if (self.isBeingObserved) {
-    dispatch_async(dispatch_get_main_queue(), ^{
-      __strong ARDynamicScreenDimensions *sself = wself;
-      if (sself && sself.isBeingObserved) {
-        [sself sendEventWithName:@"change" body:sself.currentDimensions];
-      }
-    });
-  }
+  dispatch_async(dispatch_get_main_queue(), ^{
+    __strong ARDynamicScreenDimensions *sself = wself;
+    if (!sself) {
+      return;
+    }
+    sself.currentDimensions = getScreenDimensions();
+    if (sself.isBeingObserved) {
+      [sself sendEventWithName:@"change" body:sself.currentDimensions];
+    }
+  });
```

Comment:
The documentation for this notification indicates that the `NSNotification` object actually holds the value, which seems like a less brittle approach and perhaps less blocking of the main thread https://developer.apple.com/documentation/uikit/uideviceorientationdidchangenotification?language=objc

Reply:
Or did you mean the values in the size struct?

Reply:
yeah the sizes and safe area insets were wrong too.

Reply:
Gotcha